File: src/main/java/com/gpuflight/agent/model/AgentConfig.java
package com.gpuflight.agent.model;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.gpuflight.agent.config.PublisherConfig;

import java.util.List;

public record AgentConfig(
    @JsonProperty("source") LogSourceConfig source,
    @JsonProperty("sources") List<LogSourceConfig> sources,
    @JsonProperty("publisher") PublisherConfig publisher,
    @JsonProperty("archiver") ArchiverConfig archiver
) {
    /**
     * Returns all configured sources. Supports both the legacy single
     * {@code "source"} field and the new {@code "sources"} array.
     * If both are set, they are merged (source first, then sources).
     */
    public List<LogSourceConfig> allSources() {
        var list = new java.util.ArrayList<LogSourceConfig>();
        if (source != null) list.add(source);
        if (sources != null) list.addAll(sources);
        return list;
    }
}

File: gpuflight-agent/src/main/kotlin/com/gpuflight/agent/model/AgentConfig.kt
package com.gpuflight.com.gpuflight.agent.model

import com.gpuflight.com.gpuflight.agent.config.PublisherConfig
import kotlinx.serialization.Serializable

@Serializable
data class AgentConfig(
    val source: LogSourceConfig,
    val publisher: PublisherConfig
)

@Serializable
data class LogSourceConfig(
    val folder: String = ".",
    val filePrefix: String = "gpufl"
)